src/Substrait.Core/SubstraitRelVisitor.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Substrait.Relation;
+
+namespace Substrait.Core
+{
+  /// <summary>
+  /// Visitor that has methods for accepting Substrait relations as input and provides concrete implementation to meet
+  /// its goals.
+  /// </summary>
+  public abstract class SubstraitRelVisitor<T>
```

Comment:
What is `T` here?  It's not obvious why this is needed instead of having all the methods return void.

Reply:
`void` makes sense to me. Although `substrait-java` follows a different pattern and relies on returning objects. I can fix it.